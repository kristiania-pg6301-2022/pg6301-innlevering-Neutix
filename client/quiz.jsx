import React, {useEffect, useState} from "react";

function AnswerQuestion({question, onReload}) {
    async function handleConfirm(answer) {
        const {id} = question;
        const res = await fetch("/api/question/answer", {
            method: "post",
            headers: {
                "content-type": "application/json",
            },
            body: JSON.stringify({id, answer}),
        });
        onReload();
    }

    return (
        <div>
            <h1>{question.question}</h1>
            {Object.keys(question.answers)
                .filter((a) => question.answers[a])
                .map((a) => (
                    <div key={a}>
                        <button style={{margin: "1rem", fontSize: "1.5rem"}} onClick={(q) => handleConfirm(a)}>
                            {question.answers[a]}
                        </button>
                    </div>
                ))}
        </div>
    );
}

function QuestionComponent({reload}) {
    const [question, setQuestion] = useState();

    async function handleLoadQuestion() {
        const res = await fetch("/api/question/random")
        setQuestion(await res.json());
    }

    function handleReload() {
        setQuestion(undefined)
        reload();
    }

    if (!question) {
        return (
            <div>
                <button style={{margin: "1rem", fontSize: "1.5rem"}} onClick={handleLoadQuestion}>New question</button>
            </div>
        )
    }

    return <AnswerQuestion question={question} onReload={handleReload}/>;
}

export function QuizApp() {
    const [score, setScore] = useState();
    useEffect(reload, []);

    async function reload() {
        const res = await fetch("/api/question/score");
        setScore(await res.json());
    }
    console.log(score)
    return (

        <>
            <h1>Code Quiz</h1>
            {score && (
                <div style={{fontSize: "1.5rem"}}>
                    Correct answers: {score.correct} Total answers: {score.answered}
                </div>
            )}
            <QuestionComponent reload={reload}/>
        </>
    );
}