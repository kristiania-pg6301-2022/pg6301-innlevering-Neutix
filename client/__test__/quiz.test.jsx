import ReactDOM from "react-dom";
import React from "react";
import {QuizApp} from "../quiz.jsx";

describe("Quiz test", () => {

    it("DummyTest", () =>{
        const element = document.createElement("div");
        ReactDOM.render(<QuizApp />, element);
        expect(element.innerHTML).toMatchSnapshot()
    })

    }


    )