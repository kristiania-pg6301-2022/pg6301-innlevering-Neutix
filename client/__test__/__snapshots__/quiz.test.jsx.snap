// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quiz test DummyTest 1`] = `"<h1>Code Quiz</h1><div><button style=\\"margin: 1rem; font-size: 1.5rem;\\">New question</button></div>"`;
